sdk/java-sdk-protobuf/src/main/scala/kalix/javasdk/impl/telemetry/Telemetry.scala

```diff
@@ -58,14 +58,18 @@ object Telemetry {
   }
 }
 
-class Telemetry[T](serviceName: String, system: ActorSystem, commandType: T) {
+sealed trait ComponentCategory
+case object ActionCategory extends ComponentCategory
+case object EventSourcedEntityCategory extends ComponentCategory
+
+class Telemetry(serviceName: String, system: ActorSystem, componentCategory: ComponentCategory) {
 
   import Telemetry._
 
-  val tracePrefix = commandType match {
-    case _: Command.type       => "Event Sourced Entity"
-    case _: ActionCommand.type => "Action"
-    case other                 => logger.warn("Command type not implemented [{}].", other.getClass)
+  val tracePrefix = componentCategory match {
+    case _: EventSourcedEntityCategory.type => "Event Sourced Entity"
+    case _: ActionCategory.type             => "Action"
+    case other                              => logger.warn("Command type not implemented [{}].", other.getClass)
```

Comment:
Can't be any other now since sealed. Also you could put the prefix in a method on each and not have to switch at all.

Reply:
done